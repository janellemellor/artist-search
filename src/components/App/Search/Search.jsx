import React from 'react';
import PropTypes from 'prop-types';

export const Search = ({ onSubmit, onChange, searchInput }) => (
  <>
    <form onSubmit={onSubmit}>
      <fieldset>
        <input type="text" name="artist-search" value={searchInput} onChange={onChange}  />
        <button>Find Artist!</button>
      </fieldset>
    </form>
  </>
);

Search.propTypes = {
  onSubmit: PropTypes.func.isRequired, 
  onChange: PropTypes.func.isRequired, 
  searchInput: PropTypes.string.isRequired
};

export default Search;
