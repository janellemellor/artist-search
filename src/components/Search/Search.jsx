import React from 'react';
import PropTypes from 'prop-types';

export const Search = ({ onSubmit, onChange }) => (
  <>
    <form onSubmit={onSubmit}>
      <fieldset>
        <input type="text" name="artist-search" value="name" onChange={onChange}  />
        <button>Find Artist!</button>
      </fieldset>
    </form>
  </>
);

Search.propTypes = {
  onSubmit: PropTypes.func.isRequired, 
  onChange: PropTypes.func.isRequired, 
};

export default Search;
