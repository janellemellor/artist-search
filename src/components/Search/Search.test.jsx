import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search.jsx';

describe('Search Component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Search 
      onSubmit={() => {}} onChange={() => {}} searchInput="Grouper" />);
    expect(wrapper).toMatchSnapshot();  
  });    
});
