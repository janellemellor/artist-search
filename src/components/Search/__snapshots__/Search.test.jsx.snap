// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component matches a snapshot 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <fieldset>
      <input
        name="artist-search"
        onChange={[Function]}
        type="text"
        value="Grouper"
      />
      <button>
        Find Artist!
      </button>
    </fieldset>
  </form>
</Fragment>
`;
